### ISSUE TITLE
Fix doc/examples/simple.yml

### ISSUE THREAD

**Issue by mxinden** [prometheus/alertmanager#848]
When running alertmanager with the config specified in `doc/examples/simple.yml`, alertmanager fails with 
```
ERRO[0000] Loading configuration file failed: yaml: unmarshal errors:                                     
  line 48: cannot unmarshal !!str `^(foo1|...` into map[string]config.Regexp  file="doc/examples/simple.yml" source="main.go:237"
```

If I replace `- match_re:` with `- match:` alertmanager starts without issues.

This is similarly done in `config/testdata/conf.good.yml`.

@fabxc Any idea here?

**Comment by brancz** [comment]
I'm getting a different error, but also an error unmarshaling:

```
ERRO[0000] error: yaml: unmarshal errors:
  line 26: cannot unmarshal !!map into string  source="api.go:144"
ERRO[0000] Loading configuration file failed: yaml: unmarshal errors:
  line 26: cannot unmarshal !!map into string  file="doc/examples/simple.yml" source="main.go:237"
```

Wondering why it didn't fail with the test config. Will investigate a bit.

**Comment by brancz** [comment]
Probably unrelated but regex unmarshaling is inconsistent for yaml and json https://github.com/prometheus/alertmanager/blob/d383ab0d91e44ee33c17e766300700e2a030a4ba/config/config.go#L472-L513

**Comment by mxinden** [comment]
Fixed by #850
